src/topp/DatabaseSuitability.cpp

```diff
@@ -0,0 +1,367 @@
+  ExitCodes main_(int, const char**)
+  {
+    //-------------------------------------------------------------
+    // parsing parameters
+    //-------------------------------------------------------------
+    String in_id = getStringOption_("in_id");
+    String in_spec = getStringOption_("in_spec");
+    String in_novo = getStringOption_("in_novo");
+    String out = getStringOption_("out");
+    Int novo_fract = getIntOption_("novor_fract");
+    bool no_re_rank = getFlag_("force_no_re_rank");
+
+    //-------------------------------------------------------------
+    // reading input
+    //-------------------------------------------------------------
+    IdXMLFile x;
+    vector<ProteinIdentification> prot_ids;
+    vector<PeptideIdentification> pep_ids;
+    x.load(in_id, prot_ids, pep_ids);
+
+    vector<ProteinIdentification> novo_prots;
+    vector<PeptideIdentification> novo_peps;
+    x.load(in_novo, novo_prots, novo_peps);
+
+    UInt64 count_ms2_lvl;
+    {
+      MzMLFile m;
+      PeakFileOptions op;
+      op.addMSLevel(2);
+      op.setFillData(false);
+      m.setOptions(op);
+      PeakMap exp;
+      m.load(in_spec, exp);
+      count_ms2_lvl = exp.size();
+    }
+
+    //-------------------------------------------------------------
+    // calculations
+    //-------------------------------------------------------------
+
+    // db suitability
+
+
+    double cut_off;
+    if (!no_re_rank)
+    {
+      cut_off = getDecoyCutOff_(pep_ids, novo_fract);
+      if (cut_off < 0)
+      {
+        OPENMS_LOG_ERROR << "Could not compute decoy cut off. Re-ranking impossible. If you want to ignore this, set the 'force_no_re_rank' flag." << endl;
+        return INCOMPATIBLE_INPUT_DATA;
+      }
+    }
+
+    UInt64 count_db = 0;
+    UInt64 count_novo = 0;
+    UInt64 count_re_ranked = 0;
+    UInt64 count_interest = 0;
+
+    for (const auto& pep_id : pep_ids)
+    {
+      vector<PeptideHit> hits = pep_id.getHits();
+
+      if (hits.empty()) continue;
+
+      PeptideHit top_hit = hits[0];
+
+      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
+
+      // check if top hit is found in de novo protein
+      set<String> accessions = top_hit.extractProteinAccessionsSet();
+      bool is_novo = true;
+      for (const String& acc : accessions)
+      {
+        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
+        {
+          is_novo = false;
+          break;
+        }
+      }
+
+      if (is_novo) // top hit is de novo hit
+      {
+        if (hits.size() == 1)
+        {
+          ++count_novo;
+          continue;
+        }
+
+        PeptideHit second_hit = hits[1];
+
+        if (second_hit.getMetaValue("target_decoy") == "decoy")
+        {
+          ++count_novo;
+          continue;
+        }
+
+        // check if second hit is db hit
+        set<String> second_accessions = top_hit.extractProteinAccessionsSet();
```

Comment:
this code is duplicated. Make it a function or a class, which gets passed a PeptideHit (or even better the PepID) and computes everything you want to know.